org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/CharacterLiteral.java

```diff
@@ -205,7 +205,18 @@ void internalSetEscapedValue(String value) {
 	 * @exception IllegalArgumentException if the literal value cannot be converted
 	 */
 	public char charValue() {
-		Scanner scanner = this.ast.clonedScanner();
+		// create a new local scanner to allow concurrent use
```

Comment:
better make it reusable with a org.eclipse.jdt.internal.compiler.parser.Scanner.clone()?